### ISSUE TITLE
Redux Devtools floods profiler event list

### ISSUE THREAD

**Issue by fmalcher** [mgechev/eap-feedback#2]
tl;dr; `message` events trigger CD which makes the Profiler unusable together with Redux DevTools

----

- Project with NgRx, `StoreDevToolsModule` and Redux DevTools Chrome Extension
- NgRx and Redux DevTools communicate through `window.postMessage`
- `message` events trigger Change Detection and flood the profiler event list
- the profiler is not really usable without disabling the Redux DevTools

Example project: https://github.com/angular-schule/2021-03-angular-workshop-online/tree/main/book-rating

![Screenshot 2021-03-19 at 10 27 24](https://user-images.githubusercontent.com/1683147/111759276-b30abb00-889d-11eb-806b-d9f0d49e60ab.png)

This problem is not specifically related to the Angular DevTools, but I never noticed this behavior before.
It could be solved on the NgRx or Redux DevTools side, but it would be cool to have a solution within Angular DevTools for this, too.

### Idea
- some sort of filtering for event type
- or ignoring specific events (or event sources?)

**Comment by GregOnNet** [comment]
- We experience the same issue in our Redux project.
- [Redux Devtools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?utm_source=chrome-ntp-icon) Plugin for Chrome is active
- We use [NgxSDevTools](https://www.ngxs.io/plugins/devtools)

![image](https://user-images.githubusercontent.com/444278/111797802-c29ff900-88c9-11eb-8061-6f9d3c2559b0.png)

## Observed Behaviour

- Start Profiler
- Lots of events are captured by the profiler
- The app freezes and cannot be used any more until the profiler is stoped.

**Comment by mgechev** [comment]
Looks like Redux DevTools triggers change detection cycles. Not sure if Angular DevTools should be aware of other existing tooling that communicates via `postMessage`.

So far it feels like a `won't fix`, but let me know if you have other thoughts.

**Comment by GregOnNet** [comment]
I will investigate this issue a little further.
Even without Redux DevTools the Angular DevTools recognize hundreds of CD-Cycles in one second.

Currently, I cannot tell what the root cause is.

---

> So far it feels like a won't fix, but let me know if you have other thoughts.

Maybe this question can be answered after getting more feedback how many teams out there rely on Redux DevTools. A simple solution would be to mention a workaround in the documentation.
I myself tried to disable the Redux DevTools in Chrome and in my application without success.

@fmalcher Can you try to disable Redux DevTools and see if the problem persists?

Cheers
Gregor

**Comment by fmalcher** [comment]
It got worse... 🐞 Communication with the Redux Devtools floods the event list with hundreds of entries per second.
This freezes the whole browser and I can only force quit it. It also outputs warnings on the console so I can't use the console anymore (can't even open it).

The event filter is a good means in general to go and see the CD cycles. However, for me it doesn't even come this far when I can't stop the profiling recording.

I think there are very many people using the Redux DevTools extension so we need a practical and pragmatic solution here. Otherwise those people (including me) would not be able to actually use ADT
.
What do you think about a hard switch like "Ignore window.message events"?
@mgechev 


![Screenshot 2021-04-26 at 21 02 16](https://user-images.githubusercontent.com/1683147/116137148-7a97a180-a6d3-11eb-8737-bb8515c84d6d.png)


![Screenshot 2021-04-26 at 21 07 38](https://user-images.githubusercontent.com/1683147/116137164-7ec3bf00-a6d3-11eb-8afb-33214909c2c9.png)
![Screenshot 2021-04-26 at 21 07 43](https://user-images.githubusercontent.com/1683147/116137173-81beaf80-a6d3-11eb-8a43-baf8122fd129.png)

**Comment by mgechev** [comment]
>What do you think about a hard switch like "Ignore window.message events"?

We can skip important change detection cycles this way. I'm not confident the fix should be in Angular DevTools given that these are actual change detection cycles that the app performs.
